### DISCUSSION TITLE
goto function interruption

### DISCUSSION THREAD

**Discussion by eschan146** [colobot/colobot#1673] [Ideas]
I have noticed that when I run an object's `goto` function, it runs continuously until the object reaches its destination. However, I want it to stop within sixty meters of the target to shoot it, but that means pausing the function mid-execution. Is there a way I can implement this?

Thank you.

**Comment by tomangelo2** [comment]
Get a point near destination with `space()` function. Adjust min and max parameters to suit your needs, then goto() to that point. Might not work in all cases, but this will be easiest solution.

**Comment by eschan146** [comment]
I used some vector math for this instead:
```
point direction_vector (target.position.x - position.x,
						target.position.y - position.y,
						target.position.z - position.z);
point normalized_vector = normalize(direction_vector);
point destination (target.position.x - normalized_vector.x * proximity,
					target.position.y - normalized_vector.y * proximity,
					target.position.z - normalized_vector.z * proximity);
```
and then use `flatspace` and find an area around it

**Comment by hexagonrecursion** [comment]
* @eschan146 ~I am interpreting this as a feature request rather than as a request for help on how to play the game because you posted this in [github issues](https://github.com/colobot/colobot/issues) rather than [github discussions](https://github.com/colobot/colobot/discussions) . It appears that a feature that covers the use-case you have described has already been requested.~ Related: https://github.com/colobot/colobot/issues/474
